bigquery/indexer.py

```diff
@@ -0,0 +1,114 @@
+"""Loads BigQuery table into Elasticsearch."""
+
+import csv
+import json
+import logging
+import os
+import time
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk
+import pandas as pd
+
+
+# Log to stderr.
+logging.basicConfig(level=logging.INFO,
+    format='%(asctime)s %(filename)10s:%(lineno)s %(levelname)s %(message)s',
+    datefmt='%Y%m%d%H:%M:%S')
+logger = logging.getLogger('indexer.bigquery')
+
+
+def open_and_return_json(file_path):
+  """Opens and returns JSON contents.
+
+  Args:
+    file_path: Relative path of JSON file.
+
+  Returns:
+    Parsed JSON.
+  """
+  with open(file_path,'r') as f:
+    # Remove JSON comments.
+    jsonDict = json.loads('\n'.join([row for row in f.readlines() if len(row.split('//')) == 1]))
+  return jsonDict
+
+
+def init_elasticsearch(index_name):
+  es = Elasticsearch(['http://localhost:9200'], timeout=600)
+  logger.info('Deleting and recreating %s index.' % index_name)
+  try:
+    es.indices.delete(index=index_name)
+  except Exception:
+    pass
+  es.indices.create(index=index_name,body={})
+  return es
+
+
+def index_facet_field(es, index_name, primary_key, project_id, dataset_id,
+    table_name, field_name, readable_field_name):
+  """Indexes a facet field.
+
+  Args:
+    es: Elasticsearch object.
+    index_name: Name of Elasticsearch index.
+    primary_key: Name of primary key field.
+    project_id: BigQuery project ID.
```

Comment:
If these args are hard coded anyway, could we set them as constants instead of passing them around as args?

Reply:
I think it's a bit cleaner to pass them around, see for example https://softwareengineering.stackexchange.com/a/148109